### ISSUE TITLE
Add Flask security middleware

### ISSUE THREAD

**Issue by rwblickhan** [ubclaunchpad/rocket2#224]
**Please give a one-sentence summary of the feature you would like to see.**
We should add some Flask middleware that helps with security.

**Please give as many details as possible about the requirements for the feature.**
In particular, I think [Talisman](https://github.com/GoogleCloudPlatform/flask-talisman) and [SeaSurf](https://flask-seasurf.readthedocs.io/en/latest/) would be good to add.

**Comment by rwblickhan** [comment]
Closed by #243.

**Comment by cheukyin699** [comment]
Reopened because the last time we did this, we thought that it worked fine and it didn't. The issue was with the Flask CSRF library. The error below is what we got:

```
{Time: 2019-03-08 20:32:11, Level: [WARNING], module: flask_seasurf, function: 
_before_request():249, message: Forbidden (Referer checking failed: no referer.):
 /slack/events}
```

We don't know what's wrong so we just removed the library in a hurry.

Another thing to note is that we must set `app.secret_key` to be a random string for the session to be secure or something (probably used as a CSRF token?). To the future person on the task please do so.

And remember to test the things!
